src/MainWindow.vala

```diff
@@ -250,6 +248,7 @@ namespace Terminal {
                 if (focus_timeout == 0) {
                     focus_timeout = Timeout.add (20, () => {
                         focus_timeout = 0;
+                        save_opened_terminals_with_zooms ();
```

Comment:
What is the reason for saving the tabs on `focus-in` event?

Reply:
Handle the situation when is opened first terminal, then second one, second one is closed and then is laptop rebooted. Without this code would be saved tabs and tab-zooms from second terminal but should be from first one.

Reply:
But it works only when is focus passed back to previously opened terminal, do you know what I mean, it works only when the previously opened terminal is in the layer immediately under the closed one.

Reply:
Yes, the issue of what to do about saving settings in second and subsequent windows is not really decided. One point of view (including me) is that settings only refer to the first opened window (that is what Files implements). Others think that the last *closed* window is the one that should save the settings (which is what Terminal implements).  I think the issue you are trying to fix is also present in master because rebooting forcibly kills the process and the save on close code never runs.  I would not add this line in this PR since master does not save on focus in and there is no agreed "right" answer.

Reply:
@jeremypw I am afraid that I discovered weird behavior without added resaving `tabs` and `tab-zooms` during focus in event. Because when I close terminal, then I return to previously opened terminal and I for example change cwd in any tab and then is saved broken state, because on cwd are saved only `tabs` and no `tab-zooms`, in other words saving GLib.Settings calls returns back attack like boomerang :D My added saving to focus in event would solve this problem, and in best way I think or this would be solved by not saving calls and call `tabs` or `tab-zooms` all time so when I return back to previously opened terminal and I start somehow interact with him broken state can not occurs. What do you think?

Reply:
@jeremypw I have also an idea about adding a new option continuous saving to remember tabs, so change it from bool to select with variants continuous saving, on close and do not save, on close, and could be even by default on the close variant.

Reply:
@ldrahnik Yes, on reconsideration I think we have to save on focus-in because the settings must always reflect the state of the current window.  I think the issue of whether continuous saving should be optional should be raised separately as it needs signing off by the design team to add another setting.  In general elementary tries to avoid too many settings.  There would need to be two frequently used situations where different settings are required.  Why would users not want to continuous save?

Reply:
Yes, I think you would struggle to convince the design team that was a good idea unless the app was really draining the battery with continuous saving - in which case it would be better to throttle it (Code autosaving is throttled so that it only happens after one second of inactivity).